paddle/fluid/framework/variable.h

```diff
@@ -38,6 +39,7 @@ class Variable {
 
   template <typename T>
   T* GetMutable() {
+    std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
I think we can't give var a mutex here. Vars are always not thread-safe in any programming language. You must protect your vars where you use it concurrently, but not add it here.

And, adding a mutex cannot protect the var's content actually.

Reply:
This can not make var thread-safe completely, only make `GetMutable` function thread-safe, I filed an issue to follow `make variable and gRPC client thread-safe`.

Reply:
I think we should use a workaround instead of changing Variable implement anyway.

Reply:
Maybe we can use `GRPCClient` instance as a singleton, or do you have some other ideas?